Resources/MessageQueue/StompMQConnection.py

```diff
@@ -0,0 +1,172 @@
+        self.connection.nack( headers['message-id'], self.subscriptionID )
+
+  def defaultCallback( self, headers, message ):
+    """
+    Default callback function called every time something is read from the queue
+
+    :param dict headers: headers of the MQ message
+    :param json message: json string of the message
+    """
+    dictionary = json.loads( message )
+    with self.lock:
+      self.msgList.append( dictionary )
+    return S_OK()
+
+  # Rest of functions
+
+  def setupConnection( self, parameters = None, receive = False, messageCallback = None ):
+    """
+    Establishes a new connection to a Stomp server, e.g. RabbitMQ
+
+    :param dict parameters: dictionary with additional MQ parameters if any
+    :param bool receive: flag to enable the MQ connection for getting message
+    :param func messageCallback: function to be called when a new message is received from the queue ( only receiver mode ).
+                                If None, the defaultCallback method is used instead
+    :return: S_OK/S_ERROR
+    """
+
+    self.receiver = receive
+    if self.receiver:
+      if messageCallback:
+        self.callback = messageCallback
+      elif not self.callback:
+        self.callback = self.defaultCallback
+
+    if parameters is not None:
+      self.parameters.update( parameters )
+
+    # Make the actual connection
+    host = self.parameters.get( 'Host' )
+    port = self.parameters.get( 'Port', 61613 )
+    vhost = self.parameters.get( 'VHost' )
+    user = self.parameters.get( 'User', 'guest' )
+    password = self.parameters.get( 'Password', 'guest' )
+    self.queueName = self.parameters.get( 'Queue' )
+    headers = {}
+    if "Persistent" in self.parameters and self.parameters['Persistent'].lower() in ['true', 'yes', '1']:
+      headers = { 'persistent': 'true' }
+    try:
+      self.connection = stomp.Connection( [ ( host, int( port ) ) ], vhost = vhost )
+      self.connection.start()
+      self.connection.connect( username = user, passcode = password )
+
+      if self.receiver:
+        ack = 'auto'
+        if self.parameters.get( 'Acknowledgement', '' ).lower() in ['true', 'yes', '1']:
+          self.acknowledgement = True
+          ack = 'client-individual'
+        self.subscriptionID = makeGuid()[:8]
+        self.connection.set_listener( '', self )
+        self.connection.subscribe( destination = '/queue/%s' % self.queueName,
+                                   id = self.subscriptionID,
+                                   ack = ack,
+                                   headers = headers )
+    except Exception as e:
+      return S_ERROR( EMQCONN, 'Failed to setup connection: %s' % e )
+
+    self.alive = True
+    return S_OK( 'Setup successful' )
+
+  def put( self, message ):
+    """
+    Sends a message to the queue
+    message contains the body of the message
+
+    :param str message: string or any json encodable structure
+    """
+    try:
+      if isinstance( message, list ):
```

Comment:
replace this with 
`if not isinstance( message, list ):
  message = [ message ]`
remove else branch, execute for loop unconditionally?